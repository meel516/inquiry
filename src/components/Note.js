import React from 'react';
import {Input, FormGroup, Label} from 'reactstrap';

export default function Note(props) {
  return (
    <FormGroup>
      <Label for={props.id}>{props.label}</Label>
      <Input type="textarea" name={props.name} id={props.id} />
    </FormGroup>
  )
}
