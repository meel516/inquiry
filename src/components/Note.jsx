import React from 'react'
import PropTypes from 'prop-types'
import {Input, FormGroup, Label} from 'reactstrap'

export default function Note(props) {
  return (
    <FormGroup>
      <Label for={props.id} id={props.labelId} className="label-format">{props.label}</Label>
      <Input type="textarea" name={props.name||`lead.notes.${props.id}`} id={props.id} onChange={props.onChange} onBlur={props.onBlur}/>
    </FormGroup>
  )
}

Note.propTypes = {
  label: PropTypes.string.isRequired,
  onBlur: PropTypes.func.isRequired,
  id: PropTypes.string,
}
