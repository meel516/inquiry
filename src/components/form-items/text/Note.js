import React from 'react'
import PropTypes from 'prop-types'
import { FormGroup, Label } from 'reactstrap'
import { TextArea } from '../../formik-inputs';

export const Note = ({ name, label, rows, cols, locked = false }) => {
  return (
    <FormGroup>
      <Label for={name} className="label-format">{label}</Label>
      <TextArea  name={name} cols={cols} rows={rows} disabled={locked} />
    </FormGroup>
  )
}

Note.propTypes = {
  name: PropTypes.string,
  label: PropTypes.string.isRequired,
  locked: PropTypes.bool,
  rows: PropTypes.number,
}

Note.defaultProps = {
  locked: false,
  rows: 3,
  cols: 10,
}