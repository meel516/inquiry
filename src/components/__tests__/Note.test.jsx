// Link.react.test.js
import React from 'react';
import Note from '../Note';
import renderer from 'react-test-renderer';

test('verify note rendering', () => {
    let handleChangeCalled = false,
          handleBlurCalled = false

    const handleChange = function() {
        handleChangeCalled = true;
    } 
    const handleBlur = function() {
        handleBlurCalled = true;
    }

    const component = renderer.create(
        <Note label="Additional Notes" id="additionalNotes" onChange={handleChange} onBlur={handleBlur} />
    )
    let output = component.toJSON();
    expect(output).toMatchSnapshot();
})