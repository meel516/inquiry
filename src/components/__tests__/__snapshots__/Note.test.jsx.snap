// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`verify note rendering 1`] = `
<div
  className="form-group"
>
  <label
    className="label-format"
    htmlFor="additionalNotes"
  >
    Additional Notes
  </label>
  <textarea
    className="form-control"
    id="additionalNotes"
    name="lead.notes.additionalNotes"
    onBlur={[Function]}
    onChange={[Function]}
  />
</div>
`;
